qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,385 @@
+    leakage to the |2> state.
+
+        f(x) = Gaussian + 1j * beta * d/dx [Gaussian]
+             = Gaussian + 1j * beta * (-(x - duration/2) / sigma^2) [Gaussian]
+
+    where 'Gaussian' is:
+        Gaussian(x, amp, sigma) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )
+
+    Ref:
+        [1] Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+        Analytic control methods for high-fidelity unitary operations
+        in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).
+        [2] F. Motzoi, J. M. Gambetta, P. Rebentrost, and F. K. Wilhelm
+        Phys. Rev. Lett. 103, 110501 – Published 8 September 2009
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 beta: float):
+        """Initialize the drag pulse.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Drag envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+            beta: The correction amplitude.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._beta = beta
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def beta(self):
+        return self._beta
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return drag(duration=self.duration, amp=self.amp, sigma=self.sigma,
+                    beta=self.beta, zero_ends=False)
+
+    def validate_parameters(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+        if isinstance(self.beta, complex):
+            raise PulseError("Beta must be real.")
+        # Check if beta is too large: the amplitude norm must be <=1 for all points
+        if self.beta > self.sigma:
+            # If beta <= sigma, then the maximum amplitude is at duration / 2, which is
+            # already constrainted by self.amp <= 1
+
+            # 1. Find the first maxima associated with the beta * d/dx gaussian term
+            #    This eq is derived from solving for the roots of the norm of the drag function.
+            #    There is a second maxima mirrored around the center of the pulse with the same
+            #    norm as the first, so checking the value at the first x maxima is sufficient.
+            argmax_x = (self.duration / 2
+                        - (self.sigma / self.beta) * math.sqrt(self.beta ** 2 - self.sigma ** 2))
+            if argmax_x < 0:
```

Comment:
I'd check instead that argmax_x >= 0, < duration and if that's the case do the calculation. If this isn't true, you don't need to check (since this point doesn't actually occur in the sample space).

Reply:
It is possible for the extrema at a point outside of the sample range to be so great that the point at 0 is still greater than 1